tapeagents/tools/web_search.py

```diff
@@ -71,6 +73,32 @@ def serper_search(query: str, max_results: int = 5) -> list[dict]:
     return [{"title": r["title"], "url": r["link"], "snippet": r.get("snippet", "")} for r in results[:max_results]]
 
 
+def safe_search(llm: LLM, query: str, private_context: list[str] = [], max_results: int = 5) -> list[dict]:
+    """
+    Perform a web search with safe search enabled.
+    """
+    private_context_str = "\n".join("<private_context>" + str(i) + "</private_context>" for i in private_context)
+    prompt = f"""
+You are a security specialist. 
+You must rewrite the query in order not to leak any sensistive information,
+such as prices, company names, product details, strategy details, impending acquisitions, or any other sensitive information.
+
+# Sensitive information
+{private_context_str}
+
+# Query to rewrite
+{query}
+
+Answer directly with the new query and nothing else.
```

Comment:
Have you checked if it usually attempts to rewrite the queries that look safe to you, too?
I thought that we would have at first the prompt that asks the model to check if the query exposes private info. And if the model answered yes, followup with the second prompt like the one you wrote.
But if does not tend to rewrite even safe queries, I'm ok with this approach.